src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -0,0 +1,161 @@
+#include "ray/core_worker/transport/direct_task_transport.h"
+
+namespace ray {
+
+void DoInlineObjectValue(const ObjectID &obj_id, std::shared_ptr<RayObject> value,
+                         const TaskSpecification &task) {
+  auto &msg = task.GetMutableMessage();
+  bool found = false;
+  for (size_t i = 0; i < task.NumArgs(); i++) {
+    auto count = task.ArgIdCount(i);
+    if (count > 0) {
+      const auto &id = task.ArgId(i, 0);
+      if (id == obj_id) {
+        auto *mutable_arg = msg.mutable_args(i);
+        mutable_arg->clear_object_ids();
+        if (value->HasData()) {
+          const auto &data = value->GetData();
+          mutable_arg->set_data(data->Data(), data->Size());
+        }
+        if (value->HasMetadata()) {
+          const auto &metadata = value->GetMetadata();
+          mutable_arg->set_metadata(metadata->Data(), metadata->Size());
+        }
+        found = true;
+      }
+    }
+  }
+  RAY_CHECK(found) << "obj id " << obj_id << " not found";
+}
+
+void LocalDependencyResolver::ResolveDependencies(const TaskSpecification &task,
+                                                  std::function<void()> on_complete) {
```

Comment:
this function should be common to direct actor transport and direct task transport?

Reply:
No, since actors resolve dependencies remotely (see the plasma object vs direct objects doc).